File: src/main/java/br/com/barbearia/api/controller/ServicoController.java
package br.com.barbearia.api.controller;

import br.com.barbearia.api.dto.ServicoRequestDTO;
import br.com.barbearia.api.dto.ServicoResponseDTO;
import br.com.barbearia.api.service.ServicoService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@CrossOrigin(origins = "http://localhost:8080")
@RestController
@RequestMapping("/servicos")
public class ServicoController {

    private final ServicoService service;

    public ServicoController(ServicoService service) {
        this.service = service;
    }

    @PostMapping
    public ResponseEntity<ServicoResponseDTO> criar(
            @RequestBody ServicoRequestDTO dto) {

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(service.criar(dto));
    }

    @GetMapping
    public ResponseEntity<List<ServicoResponseDTO>> listar() {
        return ResponseEntity.ok(service.listar());
    }

    @GetMapping("/{id}")
    public ResponseEntity<ServicoResponseDTO> buscarPorId(
            @PathVariable Long id) {

        return ResponseEntity.ok(service.buscarPorId(id));
    }

    @PutMapping("/{id}")
    public ResponseEntity<ServicoResponseDTO> atualizar(
            @PathVariable Long id,
            @RequestBody ServicoRequestDTO dto) {

        return ResponseEntity.ok(service.atualizar(id, dto));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deletar(@PathVariable Long id) {

        service.deletar(id);
        return ResponseEntity.noContent().build();
    }
}

File: controller/servicoController.js
import servicoService from "../services/servicoService.js";

const controller = {

  async criar(req, res) {
    const servico = await servicoService.criar(req.body);
    res.status(201).json(servico);
  },

  async listar(req, res) {
    const lista = await servicoService.listar();
    res.json(lista);
  },

  async buscarPorId(req, res) {
    const servico = await servicoService.buscarPorId(req.params.id);
    res.json(servico);
  },

  async atualizar(req, res) {
    const servico = await servicoService.atualizar(
      req.params.id,
      req.body
    );
    res.json(servico);
  },

  async deletar(req, res) {
    await servicoService.deletar(req.params.id);
    res.status(204).send();
  }

};

export default controller;